### ISSUE TITLE
Join on multi-value column does not work as expected

### ISSUE THREAD

**Issue by suneet-s** [apache/druid#9924]
### Affected Version

0.18

### Description

The following query based on the CalciteTests data does not return any rows.

```
SELECT dim3, l.v, count(*) from druid.foo f inner join lookup.lookyloo l on f.dim3 = l.k group by 1, 2
```

dim3 is a multi value column where one row has the value ['a', 'b'] and lookup has a key 'a', so this query should match one row.

**Comment by jihoonson** [comment]
@suneet-s thanks for filing this issue. This should be documented somewhere. Since join on multi-value columns is not implemented yet, I think the expectation here is undefined. It will be great if the query fails when it is a join on a multi-value column, but it will be nice if it's at least documented.

BTW, should the affected version be 0.18.x since they are the only versions which have join support?

**Comment by suneet-s** [comment]
@jihoonson - I updated the affected version to 0.18. 

I had only tested this against master, so I wasn't sure if this behavior was new in 0.19 with all the recent join enhancements

**Comment by maytasm** [comment]
I believe this should be a feature request instead of a bug. 
The relevant code is in IndexedTableJoinMatcher#makeDimensionProcessor and LookupJoinMatcher#makeDimensionProcessor. There are comments saying
` // Multi-valued rows are not handled by the join system right now; treat them as nulls.`

**Comment by maytasm** [comment]
I am +1 on failing the query and updating the docs

**Comment by suneet-s** [comment]
Fixed by #9948
